### ISSUE TITLE
Can't use Batch Variable as TRAVERSE Target

### ISSUE THREAD

**Issue by TropicalPenguin** [orientechnologies/orientdb#7452]
## Expected behavior  
Just as we can pass a record ID as the target for TRAVERSE, an equivalent batch variable should (as under **Steps to reproduce**, below) also work.

Hopefully this is related to #7435 

## Actual behavior  
The following batch executes, but it returns an empty result. Which is different to the behaviour of returning $tgt, and issuing the same TRAVERSE command on the returned record-id.

## Steps to reproduce  

Given this set-up:

```
CREATE CLASS leonardo EXTENDS V
CREATE CLASS donatello EXTENDS V
CREATE CLASS on_top_of EXTENDS E
```

The following batch returns an empty result:

```
BEGIN
LET leo = CREATE VERTEX leonardo
LET tgt = $leo
LET don = CREATE VERTEX donatello
CREATE EDGE on_top_of FROM $don TO $leo
LET leo = CREATE VERTEX leonardo
CREATE EDGE on_top_of FROM $leo TO $don
LET don = CREATE VERTEX donatello
CREATE EDGE on_top_of FROM $don TO $leo
LET leo = CREATE VERTEX leonardo
CREATE EDGE on_top_of FROM $leo TO $don
LET don = CREATE VERTEX donatello
CREATE EDGE on_top_of FROM $don TO $leo
LET leo = CREATE VERTEX leonardo
CREATE EDGE on_top_of FROM $leo TO $don
LET don = CREATE VERTEX donatello
CREATE EDGE on_top_of FROM $don TO $leo
LET leo = CREATE VERTEX leonardo
CREATE EDGE on_top_of FROM $leo TO $don
LET traversal = TRAVERSE in('on_top_of') FROM $tgt
COMMIT
RETURN $traversal
```

Even changing the final LET to this fails:

`LET traversal = TRAVERSE in('on_top_of') FROM $tgt.@rid`

**Comment by TropicalPenguin** [comment]
After a little more tinkering, it's clear that it's not so much the batch variable but the traversal itself that is failing, somehow.

If I create another class:
`CREATE CLASS lifo EXTENDS E`

and replace the three lines above COMMIT in the batch with:
```
LET top = CREATE EDGE lifo FROM $tgt TO $don
LET leo = CREATE VERTEX leonardo
CREATE EDGE on_top_of FROM $leo TO $don
UPDATE EDGE $top SET in=$leo
LET traversal = TRAVERSE in('on_top_of') FROM $tgt
```

... then the batch return value is not empty.

**Comment by luigidellaquila** [comment]
Hi @TropicalPenguin

Could you please try latest snapshot? A couple of days ago I pushed a fix that should solve this problem

Thanks

Luigi

**Comment by TropicalPenguin** [comment]
Yep, solved in 2.2.21-SNAPSHOT. Thanks!
